src/Compilers/CSharp/Test/Syntax/LexicalAndXml/RawStringLiteralLexingTests.cs

```diff
@@ -0,0 +1,284 @@
+        [InlineData("\"\"\"\n{|CS9103: |}abc\n \t\"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "")]
+        [InlineData("\"\"\"\n\t\n\"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "\t")]
+        [InlineData("\"\"\"\r\n     abc\r\n  def\r\n  \"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "   abc\r\ndef")]
+        [InlineData("\"\"\"\n \tabc\n \"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "\tabc")]
+        [InlineData("\"\"\"\r\n    abc\r\n  \"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "  abc")]
+        [InlineData("\"\"\"\r\n  abc\r\n  def\r\n  \"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "abc\r\ndef")]
+        [InlineData("\"\"\"\r\n  \"\"\r\n  \"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "\"\"")]
+        [InlineData("\"\"\"\r\n     \"abc\"\r\n     \"\"def\"\"\r\n  \"\"\"", SyntaxKind.MultiLineRawStringLiteralToken, "   \"abc\"\r\n   \"\"def\"\"")]
+        #endregion
+        public void TestSingleToken(string markup, SyntaxKind expectedKind, string expectedValue)
+        {
+            TestSingleTokenWorker(markup, expectedKind, expectedValue, leadingTrivia: false, trailingTrivia: false);
+            TestSingleTokenWorker(markup, expectedKind, expectedValue, leadingTrivia: true, trailingTrivia: false);
+
+            // If we don't have an unterminated raw string, then also try with some trailing trivia attached.
+            if (!markup.Contains("CS" + (int)ErrorCode.ERR_Unterminated_raw_string_literal))
+            {
+                TestSingleTokenWorker(markup, expectedKind, expectedValue, leadingTrivia: false, trailingTrivia: true);
+                TestSingleTokenWorker(markup, expectedKind, expectedValue, leadingTrivia: true, trailingTrivia: true);
+            }
+        }
+
+        private static void TestSingleTokenWorker(
+            string markup, SyntaxKind expectedKind, string expectedValue, bool leadingTrivia, bool trailingTrivia)
+        {
+            if (leadingTrivia)
+                markup = " /*leading*/ " + markup;
+
+            if (trailingTrivia)
+                markup += " // trailing";
+
+            MarkupTestFile.GetSpans(markup, out var input, out IDictionary<string, ImmutableArray<TextSpan>> spans);
+
+            Assert.True(spans.Count == 0 || spans.Count == 1);
+            if (spans.Count == 1)
+                Assert.True(spans.Single().Value.Length == 1);
+
+            var token = SyntaxFactory.ParseToken(input);
+            var literal = SyntaxFactory.LiteralExpression(SyntaxKind.MultiLineRawStringLiteralExpression, token);
+            token = literal.Token;
+
+            Assert.Equal(expectedKind, token.Kind());
+            Assert.Equal(input.Length, token.FullWidth);
+            Assert.Equal(input, token.ToFullString());
+            Assert.NotNull(token.Value);
+            Assert.NotNull(token.ValueText);
+            Assert.Equal(expectedValue, token.ValueText);
```

Comment:
I assume `Value` and `ValueText` hold the same string. Is that correct?

Reply:
yup.  for otehr literal kinds there is a difference.  for example, for int-literals teh .Value will be the actual Int32 value, while ValueText will be the .ToString'ed version of that.  For string literals though, it's teh same.

Reply:
(and i can update hte test to show that).

Reply:
Cool. Yes, I think adding that assertion would be good. Thanks

Reply:
added.

Reply:
I'm still not seeing a check for the value of `token.Value` (only that it's a string and not-null, lines 194 and 195)